### DISCUSSION TITLE
How to handle already sold products which are still in the cart?

### DISCUSSION THREAD

**Discussion by itsmesaskovic** [duncanmcclean/simple-commerce#841] [Help]
We are selling used books via our Simple Commerce setup.
Now we have had an issue where a user placed an order, paid for it, but it never got flagged as paid. 

Looking back in the logs, I found out that the cart was filled with a number of books for the February 10 date. However, on February 10, another visitor also added some of the selected books to the cart and fulfilled the order.
Because they were already in our first customers shopping cart, they could place an order with these books on February 22. However, upon confirmation, the order could not be processed automatically, as the books were already out of stock.

What would the best practice be to handle this before order submission / payment?

**Comment by duncanmcclean** [comment]
Ah, I think we may need to add a stock check before the customer is redirected to off-site gateways (like Mollie - which I think you're using?).

I've opened a bug report for this issue - to receive updates, you'll need to subscribe to the issue: #842

**Comment by itsmesaskovic** [comment]
You're right. We are using Mollie.
And great, thanks for picking this up!
